### PR TITLE
Expose Unity API (EMB-3301)

### PR DESCRIPTION
**Pull request by joage**
API Documentation can now be viewed at: https://embrace-io.github.io/embrace-unity-upm/api

Made using Docfx generator. https://dotnet.github.io/docfx/index.html

### CODE REVIEW

File: docfx_project/docfx.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        ],
+        "exclude": [
+          "obj/**",
+          "_site/**"
+        ]
+      }
+    ],
+    "dest": "_site",
+    "globalMetadata": {
+      "_appTitle": "DocFX website",
```
Comment (khisakuni) [inline docfx_project/docfx.json]: we probably want something else here, like "Embrace Unity Docs" or something like that.

File: docfx_project/toc.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+- name: Api Documentation
```
Comment (khisakuni) [inline docfx_project/toc.yml]: ~Api~ API

File: docfx_project/articles/intro.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# Add your introductions here!
```
Comment (khisakuni) [inline docfx_project/articles/intro.md]: Do we need this?

File: docfx_project/api/index.md [hunk 4]
```diff
@@ -0,0 +1 @@
+# Welcome to the Embrace Unity SDK
```
Comment (khisakuni) [inline docfx_project/api/index.md]: Should there be more content here?

**Review by khisakuni — COMMENTED** [review]
Few comments. I'm not sure what the original spec for this project is so can't really comment on if it fulfills all requirements. Should Fredric take a look instead since he's involved in Unity? Or looks like EZE is back tomorrow, so should he take a look then?

File: docfx_project/articles/intro.md (hunk 3 above)
Comment (joage) [reply docfx_project/articles/intro.md]: nope

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    name: A job to run docfx
```
Comment (fnewberg) [inline .github/workflows/main.yml:20]: nitpick:  should this be something like `Build and publish Unity API docs`?

File: docfx_project/docfx.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
+{
```
Comment (fnewberg) [inline docfx_project/docfx.json:1]: 1. we should probably exclude `Embrace_Tests`. i would imagine there's some setting to do that
2. none of the methods have any description as to what they do. do we need to add comments to the source to get this or where is that supposed to come from? if adding codes is the answer, please add comments in this PR or open a separate PR for that. you can get the descriptions for most (all?) methods from the [android API docs](https://embrace-io.github.io/embrace-android-sdk3/io/embrace/android/embracesdk/Embrace.html)

Comment (joage) [reply docfx_project/docfx.json:1]: 1. can do.
2. Yes we can add comments. I'll keep the same PR. Thanks for pointing this out!

File: .github/workflows/main.yml (hunk 5 above)
Comment (joage) [reply .github/workflows/main.yml:20]: yea probably

**Review by elanz-embrace — APPROVED** [review]
This is a great way to get things published, approved from that perspective.  This doesn't have the header comments to really explain what each method does though, for the docs to be useful we'll need to add that in a follow-up.

**Comment by joage** [comment]
> This is a great way to get things published, approved from that perspective. This doesn't have the header comments to really explain what each method does though, for the docs to be useful we'll need to add that in a follow-up.

Thanks, eze! Yes the header comments are on the way. Going to merge this now, and include method summaries, parameter and return type descriptions in a later PR. @fnewberg Because of an urgent Unity fix, i won't get a chance to work on this until next week.

**Pull request merged**